pkg/resource/deploy/step_generator.go

```diff
@@ -728,10 +734,18 @@ func (sg *stepGenerator) continueStepsFromRefresh(event ContinueResourceRefreshE
 	if goal.RetainOnDelete != nil {
 		retainOnDelete = *goal.RetainOnDelete
 	}
+
+	// Carry the refreshBeforeUpdate flag forward if present in the old state.
+	refreshBeforeUpdate := false
```

Comment:
Nit: might be neater to copy the style above and just `var refreshBeforeUpdate bool` here, but obviously not important and can be ignored.